### PR TITLE
fix: correct alignment extension for bowtie2/align

### PR DESCRIPTION
**Pull request by matq007**
In the currect setup, when setting samtools to view, a `bam` file is generated instead of `sam`.

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/bowtie2/align/main.nf [hunk 1]
```diff
@@ -38,6 +38,7 @@ process BOWTIE2_ALIGN {
     }
 
     def samtools_command = sort_bam ? 'sort' : 'view'
+    def extension = samtools_command == 'sort' ? 'bam' : 'sam'
```
Comment (nvnieuwk) [inline modules/nf-core/bowtie2/align/main.nf:41]: Can you get the extension from the `args2` instead? `samtools view` and `samtools sort` both have the `--output-fmt SAM|BAM|CRAM` option from which you can easily define what the output format should be

Comment (matq007) [reply modules/nf-core/bowtie2/align/main.nf:41]: Ah, I completely missed this option. My bad, of course, this makes more sense. Closing the PR then.

**Comment by matq007** [comment]
As pointed out by @nvnieuwk, `args2` can contain `--output-fmt`.

**Comment by matq007** [comment]
I wrote a regex trick, tested it on internal pipeline, was able to generate SAM files after setting `ext.args2 = "--output-fmt SAM"`

File: modules/nf-core/bowtie2/align/main.nf [hunk 2]
```diff
@@ -38,6 +38,7 @@ process BOWTIE2_ALIGN {
     }
 
     def samtools_command = sort_bam ? 'sort' : 'view'
+    def extension = args2.contains("--output-fmt") ? (args2 =~ /--output-fmt+\s+(\S+)/)[0][1].toLowerCase() : ""
```
Comment (nvnieuwk) [inline modules/nf-core/bowtie2/align/main.nf:41]: `--output-fmt` also has a short form `-O` can you also make sure this accounts for that too?

**Review by nvnieuwk — COMMENTED** [review]
Can you add some tests for the different file output types? (preferable where you mix the use of `--output-fmt` and `-O`)

**Comment by matq007** [comment]
And we're ready

```bash
==================================================== 94 passed, 1651 skipped in 189.09s (0:03:09) =====================================================
```

**Review by nvnieuwk — DISMISSED** [review]
Awesome! Can you just fix the linting fails? Then you can merge!

**Comment by matq007** [comment]
There is only 1 linting issue for `semibin/singleeasybin`. Shouldn't that be fixed with another PR? Not relevant for this PR, I haven't touched those files.

**Comment by nvnieuwk** [comment]
> There is only 1 linting issue for `semibin/singleeasybin`. Shouldn't that be fixed with another PR? Not relevant for this PR, I haven't touched those files.

It's an easy fix (it's just asking for more keywords in the meta.yml)

**Comment by nvnieuwk** [comment]
Tests are still failing can you rerender the tests for those modules/subworkflows? We try not to have failing modules/subworkflows on the main branch

**Comment by matq007** [comment]
I see where the problem is. The alignment workflow always produces different `md5sum`, seems like there is always some slight deviation in the file. Can I remove the md5 checksum then? I know it doesn't sounds ideal, but as long you generate the aligned file it should be fine right?

**Comment by nvnieuwk** [comment]
> I see where the problem is. The alignment workflow always produces different `md5sum`, seems like there is always some slight deviation in the file. Can I remove the md5 checksum then? I know it doesn't sounds ideal, but as long you generate the aligned file it should be fine right?

Yes that should be fine! Can you also add a 'contains' section to the output files where that's possible?

**Review by nvnieuwk — APPROVED** [review]
Awesome! Thank you for this! You can merge when the tests pass :)

**Comment by matq007** [comment]
`singleeasybin` is not failing which was ok before. Can I merge anyway?

**Comment by nvnieuwk** [comment]
Yeah go ahead! Conda tests usually differ in md5sum, it's fine as long as docker and singularity pass

**Pull request merged**